tests/test_interface.sh

```diff
@@ -80,5 +80,89 @@ if output=$(TESTING_SKIP_ROOT_CHECK=1 "$CC_SNAPSHOT" -d mytest 2>&1); then
 else
   fail "Dry-run exited with error: $output"
 fi
+# Srart of custom source path tests
+
+export CC_SNAPSHOT_TAR_PATH="${PWD}/snapshot.tar"
+DEFAULT_TAR="${PWD}/snapshot.tar"
+TESTDIR="/tmp/cc_test"
+
+rm -f "$DEFAULT_TAR"
+rm -rf "$TESTDIR"
+
+mkdir -p "$TESTDIR/src/subdir"
+echo hello > "$TESTDIR/src/file1.txt"
+echo world > "$TESTDIR/src/subdir/file2.txt"
+echo remove_me > "$TESTDIR/src/remove_me.txt"
+
+#Test 5: basic directory snapshot
+if output=$(TESTING_SKIP_ROOT_CHECK=1 "$CC_SNAPSHOT" -u -s "$TESTDIR/src" mytest 2>&1); then
+  pass "ran basic directory snapshot"
+else
+  echo "$output"
+  fail "cc-snapshot failed for basic snapshot"
+fi
+
+# Verify directory created
+if [[ "$output" != *"Tarball of $TESTDIR/src is ready at $DEFAULT_TAR"* ]]; then
+  fail "Missing success message for $DEFAULT_TAR"
+fi
+
+# Verify the tarball exists
+if [[ ! -s "$DEFAULT_TAR" ]]; then
+  fail "Expected tarball $DEFAULT_TAR to exist and be non-zero size"
+fi
+pass "Basic tarball created at $DEFAULT_TAR"
+
+# Compare all contents (including removed file)
+diff \
+  <(cd "$TESTDIR/src" && find . -mindepth 1 | sed 's|^\./||; s|/$||' | sort) \
+  <(tar -tf "$DEFAULT_TAR" | sed 's|^\./||; s|/$||' | grep -v '^$' | sort) \
+  || fail "Contents mismatch for basic snapshot"
```

Comment:
this looks hard to follow. is there a more straightforward way we can compare the contents? and/or maybe we need some comments explaining how this works